app/components/modules/lunchgroup/GroupDetails.tsx

```diff
@@ -0,0 +1,111 @@
+import { LunchGroupProps } from '@typing/props';
+import { SanitizedUsers, GetUserProfilesResponse } from '@typing/api';
+import Image from 'next/image';
+import AttendeesList from '../dashboard/components/AttendeesList';
+import { useLunchGroup } from '@hooks/LunchGroupContext';
+import { OneToManyRelationships } from '@typing/types';
+import { useEffect, useState } from 'react';
+
+const GroupDetails: React.FC<LunchGroupProps> = ({ group }) => {
+  const { name, start, end, creator = {}, foods = [], users = {} } = group;
+  const { getProfiles } = useLunchGroup();
+  const [owner, setOwner] = useState('');
+  const [ownerPhotoURL, setOwnerPhotoURL] = useState('');
+  const [profiles, setProfiles] = useState<SanitizedUsers>([]);
+
+  const getUserProfiles = (
+    users: OneToManyRelationships,
+    getProfiles: CallableFunction,
+    setProfiles: CallableFunction
+  ): void => {
+    getProfiles(Object.keys(users)).then((users: GetUserProfilesResponse) =>
+      setProfiles(users.profiles || [])
+    );
+  };
+
+  const getCreatorProfile = (
+    creator: OneToManyRelationships,
+    getProfiles: CallableFunction,
+    setOwner: CallableFunction
+  ): void => {
+    getProfiles(Object.keys(creator)).then((users: GetUserProfilesResponse) =>
+      setOwner((users.profiles && 'email' in users.profiles[0] && users.profiles[0].email) || '')
+    );
+  };
+
+  useEffect(() => {
+    getCreatorProfile(creator, getProfiles, setOwner), [creator];
+    getCreatorImage();
+  });
+  useEffect(() => getUserProfiles(users, getProfiles, setProfiles), [users]);
+
+  const getFoodItems = () => {
+    const food = Object.keys(foods).map((foodItems) => foodItems);
+    return food.length === 0 ? (
+      <span>
+        <span className="text-xl">No foods selected - here are some ideas:</span>
+        <div className="text-md text-gray-500 pt-3"> Pizza, Vegan, Japanese, and Cuban</div>
+      </span>
+    ) : (
+      food.map((foodItem) => foodItem)
+    );
+  };
```

Comment:
This seems like it could be its own sub-component probably

Reply:
Definitely 👌